### PR TITLE
Use delta_sharing_scope instead of delta_sharing_enabled for delta sharing enablement

### PR DESCRIPTION
**Pull request by wchau**

### CODE REVIEW

File: catalog/resource_metastore.go [hunk 1]
```diff
@@ -76,8 +76,8 @@ func ResourceMetastore() *schema.Resource {
 				Type:     schema.TypeBool,
 				Optional: true,
 			}
-			m["delta_sharing_enabled"].RequiredWith = []string{"delta_sharing_recipient_token_lifetime_in_seconds"}
-			m["delta_sharing_recipient_token_lifetime_in_seconds"].RequiredWith = []string{"delta_sharing_enabled"}
+			m["delta_sharing_scope"].RequiredWith = []string{"delta_sharing_recipient_token_lifetime_in_seconds"}
```
Comment (alexott) [inline catalog/resource_metastore.go:79]: Can you add validation for the possible values of `delta_sharing_scope` using the `m["delta_sharing_scope"].ValidateFunc = validation.StringInSlice([]string{...})` ?

Comment (wchau) [reply catalog/resource_metastore.go:80]: Done

File: catalog/resource_metastore.go [hunk 2]
```diff
@@ -7,6 +7,7 @@ import (
 
 	"github.com/databrickslabs/terraform-provider-databricks/common"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+  "github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
```
Comment (nfx) [inline catalog/resource_metastore.go:10]: please run `make fmt`

**Review by nfx — CHANGES_REQUESTED** [review]
`make fmt` :)

File: catalog/resource_metastore.go (hunk 2 above)
Comment (wchau) [reply catalog/resource_metastore.go]: Done

**Review by nfx — APPROVED** [review]

**Pull request merged**